docs: Add proxy configuration instructions

abernix opened the pull request:
> Note: This PR currently doesn't link this new content from anywhere on the sidebar.  Furthermore, it has created the file at the top-level of the repository.  I am specifically looking for guidance on where it should live!  (It's entirely possible that it should live within the Platform documentation, though I could see a future version of Apollo Server - which theoretically had more granular proxy configuration - introducing the need to add additional `agent` documentation to this (Apollo Server) repository itself.

When utilizing Apollo Platform functionality which requires talking to the Apollo Graph Manager (formerly, Apollo Engine), it is sometimes necessary to define a proxy (often, corporately mandated) to permit requests to Apollo infrastructure (e.g. to fetch managed federation configuration and operation registry manifests).

This documentation demonstrates how to configure the [`global-agent`][Ref 1] package in order to configure such functionality.  While the `global-agent` package only supports Node.js v10+, this should be sufficient for most cases and is a clean approach which doesn't monkeypatch internals, as was
necessary in pre-v10 versions.

[Ref 1]: https://npm.im/global-agent


<!--
First, 🌠 thank you 🌠 for taking the time to consider a contribution to Apollo!

Here are some important details to follow:

* ⏰ Your time is important
          To save your precious time, if the contribution you are making will take more
          than an hour, please make sure it has been discussed in an issue first.
          This is especially true for feature requests!
* 💡 Features
          Feature requests can be created and discussed within a GitHub Issue.  Be
          sure to search for existing feature requests (and related issues!) prior to
          opening a new request.  If an existing issue covers the need, please upvote
          that issue by using the 👍 emote, rather than opening a new issue.
* 🔌 Integrations
          Apollo Server has many web-framework integrations including Express, Koa,
          Hapi and more.  When adding a new feature, or fixing a bug, please take a
          peak and see if other integrations are also affected.  In most cases, the
          fix can be applied to the other frameworks as well.  Please note that,
          since new web-frameworks have a high maintenance cost, pull-requests for
          new web-frameworks should be discussed with a project maintainer first.
* 🕷 Bug fixes
          These can be created and discussed in this repository. When fixing a bug,
          please _try_ to add a test which verifies the fix.  If you cannot, you should
          still submit the PR but we may still ask you (and help you!) to create a test.
* 📖 Contribution guidelines
          Follow https://github.com/apollographql/apollo-server/blob/master/CONTRIBUTING.md
          when submitting a pull request.  Make sure existing tests still pass, and add
          tests for all new behavior.
* ✏️ Explain your pull request
          Describe the big picture of your changes here to communicate to what your
          pull request is meant to accomplish.  Provide 🔗 links 🔗 to associated issues!

We hope you will find this to be a positive experience!  Open source contribution can be intimidating and we hope to alleviate that pain as much as possible.  Without following these guidelines, you may be missing context that can help you succeed with your contribution, which is why we encourage discussion first.  Ultimately, there is no guarantee that we will be able to merge your pull-request, but by following these guidelines we can try to avoid disappointment.
-->

jhampton commented:
Thanks for putting this together @abernix !!!

My vote is that references be added [here](https://www.apollographql.com/docs/graph-manager/federation/#validating-changes-to-the-graph) and [here](https://www.apollographql.com/docs/graph-manager/federation/#integrating-with-your-deployment-pipeline) to...

A new entry on the sidebar that reads something like "Graph Manager Network Connections"

This section can/should also contain the IP's that should be whitelisted and can be expanded with troubleshooting steps for networking environments that are more restrictive or have MITM's like proxy servers or other stateful inspection between the CLI and our cloud services.

Thanks again!

abernix commented:
@StephenBarlow I've updated this PR to rename the _Migration_ section to _Appendices_ and added this new document to it.  I haven't updated the actual file structure on disk, though the existing migration documents were already unanchored in a directory, so not too much different there.  In fact, aside from the _API References_ section above this _Appendices_ being also an appendix-like-concept, I think I actually enjoy not having a major section on the sidebar named "Migrations".  Let's iterate on this more as necessary, but since this document is in regular need by current users, it seems like getting it publicized sooner rather than later is key.

#### Before

<img src="https://user-images.githubusercontent.com/841294/66126844-05e1ab80-e5f3-11e9-8e38-e33322a14e09.png" width=200>

#### After

<img src="https://user-images.githubusercontent.com/841294/66126759-d92d9400-e5f2-11e9-9336-612e765222ba.png" width=200>
